### ISSUE TITLE
Date typing information from arrow lost after sample.

### ISSUE THREAD

**Issue by bmschmidt** [uwdata/arquero#99]
Thanks again for this great library.

If you load a table using aq.fromArrow, date type information is preserved;
but after some operations, the dates revert to being represented as some kind of integers, not as datetime.

Apologies for not providing an example inline, because I'm not confident about generating
an Arrow date field in Javascript. In the notebook at https://observablehq.com/@bmschmidt/a-files,
you can see that date information is preserved after a filter

```javascript
table.filter(d => d.country=="Mexico").select("birth date").view()
```

gives 

```
birth date
--
1915-06-21T00:00:00.000Z
1900-04-14T00:00:00.000Z
1899-04-15T00:00:00.000Z
1886-04-12T00:00:00.000Z

```

But after a sample or slice, you get ints:

```javascript
table.sample(10).select("birth date").view()
```

gives 

```
birth date
--
-1841614848
1234978816
-275112960
689860608


```

**Comment by bmschmidt** [comment]
I should clarify that this situation seems to hold with `{unpack: true}` and `{unpack: false}`.

I'm also confused about what these numbers mean -- I would have thought milliseconds since Unix epoch which arrow uses as backing, but they are not. (All of the people in that particular collection were born before 1920, but half the dates are positive.)

**Comment by jheer** [comment]
Thanks! I *think* this is due to a reify bug I just fixed today over in #97. I will add some additional tests to confirm.
